pkg/datapath/loader/tc.go

```diff
@@ -25,9 +25,17 @@ func attachSKBProgram(device netlink.Link, prog *ebpf.Program, progName, bpffsDi
 	}
 
 	if tcxEnabled {
-		// Attach using tcx if available. This is seamless on interfaces with
-		// existing tc programs since attaching tcx disables legacy tc evaluation.
-		err := upsertTCXProgram(device, prog, progName, bpffsDir, parent)
+		var err error
+		if device.Type() == "netkit" {
+			// If the device is a netkit device, we know that netkit links are
+			// supported, therefore use netkit instead of tcx. For all others like
+			// host devices, rely on tcx.
+			err = upsertNetkitProgram(device, prog, progName, bpffsDir, parent)
+		} else {
+			// Attach using tcx if available. This is seamless on interfaces with
+			// existing tc programs since attaching tcx disables legacy tc evaluation.
+			err = upsertTCXProgram(device, prog, progName, bpffsDir, parent)
+		}
 		if err == nil {
 			// Created tcx link, clean up any leftover legacy tc attachments.
 			if err := removeTCFilters(device, parent); err != nil {
```

Comment:
I'm curious: Could one hypothetically attach legacy tc via netlink to a netkit device? Or does this function fail for netkit devices but the tests don't complain because we just emit a warning?

Reply:
This would be possible, but when someone selects netkit the code enforces tcx enablement.